resilience4j-spring-cloud2/src/main/java/io/github/resilience4j/bulkhead/autoconfigure/RefreshScopedBulkheadAutoConfiguration.java

```diff
@@ -1,19 +1,78 @@
+import io.github.resilience4j.common.bulkhead.configuration.ThreadPoolBulkheadConfigurationProperties;
+import io.github.resilience4j.consumer.EventConsumerRegistry;
+import io.github.resilience4j.core.registry.RegistryEventConsumer;
+import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.boot.autoconfigure.AutoConfigureAfter;
 import org.springframework.boot.autoconfigure.AutoConfigureBefore;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
 import org.springframework.cloud.autoconfigure.RefreshAutoConfiguration;
 import org.springframework.cloud.context.scope.refresh.RefreshScope;
+import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 
 
 @Configuration
 @ConditionalOnClass({Bulkhead.class, RefreshScope.class})
 @AutoConfigureAfter(RefreshAutoConfiguration.class)
 @AutoConfigureBefore(BulkheadAutoConfiguration.class)
-public class RefreshScopedBulkheadAutoConfiguration extends
-    AbstractRefreshScopedBulkheadConfiguration {
+public class RefreshScopedBulkheadAutoConfiguration {
+    protected final BulkheadConfiguration bulkheadConfiguration;
+    protected final ThreadPoolBulkheadConfiguration threadPoolBulkheadConfiguration;
 
+    protected RefreshScopedBulkheadAutoConfiguration() {
+        this.threadPoolBulkheadConfiguration = new ThreadPoolBulkheadConfiguration();
+        this.bulkheadConfiguration = new BulkheadConfiguration();
+    }
+
+    /**
+     * @param bulkheadConfigurationProperties bulkhead spring configuration properties
+     * @param bulkheadEventConsumerRegistry   the bulkhead event consumer registry
+     * @return the RefreshScoped BulkheadRegistry
+     */
+    @Bean
+    @org.springframework.cloud.context.config.annotation.RefreshScope
+    @ConditionalOnMissingBean
+    public BulkheadRegistry bulkheadRegistry(
+        BulkheadConfigurationProperties bulkheadConfigurationProperties,
+        EventConsumerRegistry<BulkheadEvent> bulkheadEventConsumerRegistry,
+        RegistryEventConsumer<Bulkhead> bulkheadRegistryEventConsumer,
+        @Qualifier("compositeBulkheadCustomizer") CompositeCustomizer<BulkheadConfigCustomizer> compositeBulkheadCustomizer) {
+        return bulkheadConfiguration
+            .bulkheadRegistry(bulkheadConfigurationProperties, bulkheadEventConsumerRegistry,
+                bulkheadRegistryEventConsumer, compositeBulkheadCustomizer);
+    }
+
+    /**
+     * @param threadPoolBulkheadConfigurationProperties thread pool bulkhead spring configuration
+     *                                                  properties
+     * @param bulkheadEventConsumerRegistry             the bulk head event consumer registry
+     * @return the RefreshScoped ThreadPoolBulkheadRegistry
+     */
+    @Bean
+    @org.springframework.cloud.context.config.annotation.RefreshScope
```

Comment:
```suggestion
    @RefreshScope
```
and Please import RefreshScope. Others are same too.

Reply:
I can not as there are 2 RefreshScope in the same class , one for the context imple and one for the annotation so u need to differentiate between them otherwise u will have import conflict

Reply:
https://github.com/resilience4j/resilience4j/blob/577025d55f66854b0b657cf9dfb0486686294332/resilience4j-spring-cloud2/src/main/java/io/github/resilience4j/bulkhead/autoconfigure/RefreshScopedBulkheadAutoConfiguration.java#L13
Oh You mean this line, right?  I missed it.